404-fallback.php

```diff
@@ -17,20 +17,26 @@
 require_once plugin_dir_path( __FILE__ ) . '/lib/menu-site.php';
 
 function fb404_redirect_404() {
-	global $wp;
-
-	$request = wp_unslash( $_SERVER['REQUEST_URI'] );
-	$url = stripslashes( get_option( 'fb404_setting_fallback_url' ));
-	if ( is_404() && !empty($url) ) {
-		$location      =  $url . $request;
-		$status        = 302;
-		$x_redirect_by = '404 Fallback';
-		wp_redirect( $location, $status, $x_redirect_by );
-		die();
+	$request = fb404_get_request();
+
+	$url = fb404_validate_url( get_option( 'fb404_setting_fallback_url', '' ) );
+	if ( empty( $url ) ) {
+		return;
 	}
-}
 
-add_action( 'template_redirect', 'fb404_redirect_404' );
+	/**
+	 * We make sure this URL has a trailing slash, as we removed it from the Request.
+	 */
+	$location  = trailingslashit( $url);
+	$location .= $request;
+
+	$status        = 302;
+	$x_redirect_by = '404 Fallback';
+
+	wp_redirect( $location, $status, $x_redirect_by );
```

Comment:
Just noting; I'd be tempted to wrap `$location` in an `apply_filters()` to enable some customisation.

Main use case I can think of is wanting to add like `utm_source` for analytics and logs on the receiver. Could be useful for some to be able to break down how many requests are coming through the Fallback URL configuration.